cluster/src/main/java/org/apache/iotdb/cluster/client/async/AsyncClientPool.java

```diff
@@ -195,17 +195,17 @@ void onComplete(Node node) {
 
   void recreateClient(Node node) {
     ClusterNode clusterNode = new ClusterNode(node);
-    synchronized (this) {
-      Deque<AsyncClient> clientStack =
-          clientCaches.computeIfAbsent(clusterNode, n -> new ArrayDeque<>());
+    Deque<AsyncClient> clientStack =
+        clientCaches.computeIfAbsent(clusterNode, n -> new ArrayDeque<>());
+    synchronized (clientStack) {
       try {
         AsyncClient asyncClient = asyncClientFactory.getAsyncClient(node, this);
         clientStack.push(asyncClient);
       } catch (IOException e) {
         logger.error("Cannot create a new client for {}", node, e);
         nodeClientNumMap.computeIfPresent(clusterNode, (n, cnt) -> cnt - 1);
       }
-      this.notifyAll();
+      clientStack.notifyAll();
```

Comment:
hi ，this is good job.   
this is a minor issue .  "this.notifyAll()" is  related "this.wait()".   
please see 126 line.

Reply:
fixed

Reply:
LGTM